Xamarin.Forms.Platform.iOS/Platform.cs

```diff
@@ -247,9 +260,7 @@ Task INavigation.PushModalAsync(Page modal, bool animated)
 
 			modal.DescendantRemoved += HandleChildRemoved;
 
-			if (_appeared)
-				return PresentModal(modal, _animateModals && animated);
-			return Task.FromResult<object>(null);
+			return PresentModal(modal, _animateModals && animated);
```

Comment:
`PushModalAsync` should not return a `Task` with no result. This is going to abort push in cases where `_appeared` is false. We'll just remove the check and call `PresentModal` right away.